src/components/SideBar/index.tsx

```diff
@@ -0,0 +1,171 @@
+'use client'
+
+import Link from "next/link";
+import Image from "next/image";
+import { usePathname, useRouter, useSearchParams } from 'next/navigation'
+import { useState } from "react";
+import {
+  ArrowRightFromLine,
+  DiscAlbum,
+  Flag,
+  Gauge,
+  Globe,
+  Guitar,
+  Home,
+  Library,
+  Moon,
+  Settings,
+  Sun,
+  User,
+  Users
+} from "lucide-react";
+import { parseClassName } from "@/utils/helpers";
+import SideBarHeader from "./SideBarHeader";
+import styles from "./sidebar.module.css";
+
+const navigation = [
+  {
+    links: [
+      {
+        href: "/home",
+        name: "Home",
+        icon: Home
+      },
+      {
+        href: "/community",
+        name: "Community",
+        icon: Globe,
+      },
+      {
+        href: "/dashboard",
+        name: "Dashboard",
+        icon: Gauge,
+      }
+    ]
+  },
+  {
+    separator: "Leaderboards",
+    links: [
+      {
+        href: "/player",
+        name: "Player",
+        icon: User
+      },
+      {
+        href: "/band",
+        name: "Band",
+        icon: Users,
+      },
+      {
+        href: "/instrument",
+        name: "Instrument",
+        icon: Guitar,
+      },
+      {
+        href: "/country",
+        name: "Country",
+        icon: Flag,
+      },
+      {
+        href: "/setlist",
+        name: "Setlist",
+        icon: Library,
+      },
+      {
+        href: "/source",
+        name: "Source",
+        icon: DiscAlbum,
+      }
+    ]
+  },
+  {
+    separator: true,
+    links: [
+      {
+        href: "/settings",
+        name: "Settings",
+        icon: Settings
+      }
+    ]
+  }
+]
+
+const languages = [
+  { name: "English", acronym: "en"},
+  { name: "Português", acronym: "pt"}
+]
+
+const release = {
+  image: "https://news.yarg.in/images/banners/v0-12-release.webp",
+  target: "#!",
+  text: "New release 0.12" 
+}
+
+const version = "v0.12.0"
+
+const SideBar: React.FC = () => {
+  const router = useRouter();
+  const query = useSearchParams();
+  const isClose = query.has('closed')
+  const [theme, setTheme] = useState<'dark'|'light'>('dark')
+  const pathname = usePathname()
+
+  const handleTheme = () => setTheme((current) => current === 'dark' ? 'light': 'dark')
+
+  const handleToggle = () => 
+    router.push(`${pathname}${!isClose ? "?closed" : "?"}`)
+  
+  return (
+    <div className={styles.sidebar} aria-expanded={!isClose}>
+      {!isClose ? <SideBarHeader.Open tag={version} onClose={handleToggle}/> : <SideBarHeader.Close />}
+      <div className={styles.navigation}>
+        {isClose && (
+          <button type="button" className={styles.link} onClick={handleToggle}>
+            <ArrowRightFromLine className={styles.icon} />
+          </button>
+        )}
+        {navigation?.map(({separator, links}) => (
+          <section key={crypto.randomUUID()} className={styles.section}>
```

Comment:
on this .map the `key` really need to be cryptographically secure? can you explain? 

if there is no need to use crypto and you just need a key for each item (like 0, 1, 2, 3, 4) you can try this:
`{navigation?.map(({separator, links}, idx: number)`
- idx will be the index for each item.

but like a said, if it's necessary to use Crypto, that's good to me.